lib/mix/lib/mix/tasks/test.ex

```diff
@@ -569,6 +575,34 @@ defmodule Mix.Tasks.Test do
     end
   end
 
+  defp validate_max_failures_opts(opts) do
```

Comment:
What do you think about using `Integer.parse/2` to avoid control-flow via exceptions?

```elixir
defp validate_max_failures_opts(opts) do
  case Keyword.fetch(opts, :max_failures) do
    {:ok, value} when value not in ["infinity", ":infinity"] ->
      case Integer.parse(value) do
        {max_failures, _} when max_failures >= 1 -> Keyword.put(opts, :max_failures, :infinity)
        :error -> raise "--max-failuers : Expected type positive integer or infinity, got '#{value}'"
      end
      
    _ -> 
      Keyword.put(opts, :max_failures, :infinity)
  end
end
```

Additionally, it covers:

* the edge case of `--max-failures 0`
* invalid values like `--max-failures invalid`. Similar to what other flags behave:

    ```
    $ mix test --slowest a
    ** (Mix) Could not invoke task "test": 1 error found!
    --slowest : Expected type integer, got "a"
    ```

Reply:
looks greate. i renamed it `normalize_max_failures`, and I changed a bit the logic of your code to deal with invalid decimal numbers
$ mix test --max-failures 0
** (RuntimeError) --max-failures : Expected type positive integer or infinity, got '0'